signer/tests/integration/setup.rs

```diff
@@ -0,0 +1,351 @@
+    pub async fn store_deposit_request(&self, db: &PgStore) {
+        let deposit_request = model::DepositRequest {
+            txid: self.deposit_request.outpoint.txid.into(),
+            output_index: self.deposit_request.outpoint.vout,
+            spend_script: self.deposit_request.deposit_script.clone().into(),
+            reclaim_script: self.deposit_request.reclaim_script.clone().into(),
+            recipient: self.deposit_recipient.clone().into(),
+            amount: self.deposit_request.amount,
+            max_fee: self.deposit_request.max_fee,
+            sender_script_pub_keys: Vec::new(),
+        };
+
+        db.write_deposit_request(&deposit_request).await.unwrap();
+    }
+
+    /// Store how the signers voted on the deposit request.
+    ///
+    /// The deposit request must be stored in the database before this
+    /// function is called.
+    ///
+    /// This function uses the `self.deposit_request.signer_bitmap` field
+    /// to generate the cooresponding deposit signer votes and then stores
+    /// these decisions in the database.
+    pub async fn store_deposit_decisions(&self, db: &PgStore) {
+        let deposit_signers = self
+            .signer_keys
+            .iter()
+            .copied()
+            .zip(self.deposit_request.signer_bitmap)
+            .map(|(signer_pub_key, is_rejected)| model::DepositSigner {
+                txid: self.deposit_request.outpoint.txid.into(),
+                output_index: self.deposit_request.outpoint.vout,
+                signer_pub_key,
+                is_accepted: !is_rejected,
+            });
+
+        for decision in deposit_signers {
+            db.write_deposit_signer_decision(&decision).await.unwrap();
+        }
+    }
+
+    /// Use the bitmap in the `self.withdrawal_request.signer_bitmap` field to
+    /// generate the cooresponding deposit signer votes and store these
+    /// decisions in the database.
+    pub async fn store_withdrawal_decisions(&self, db: &PgStore) {
+        let withdrawal_signers: Vec<model::WithdrawalSigner> = self
+            .signer_keys
+            .iter()
+            .copied()
+            .zip(self.deposit_request.signer_bitmap)
+            .map(|(signer_pub_key, is_rejected)| model::WithdrawalSigner {
+                request_id: self.withdrawal_request.request_id,
+                block_hash: self.withdrawal_request.block_hash,
+                txid: self.withdrawal_request.txid,
+                signer_pub_key,
+                is_accepted: !is_rejected,
+            })
+            .collect();
+
+        for decision in withdrawal_signers {
+            db.write_withdrawal_signer_decision(&decision)
+                .await
+                .unwrap();
+        }
+    }
+
+    pub async fn store_withdrawal_request(&self, db: &PgStore) {
+        let block = model::StacksBlock {
+            block_hash: self.withdrawal_request.block_hash,
+            block_height: self.sweep_block_height,
```

Comment:
nit: I guess is not really important here, but isn't `sweep_block_height` related to the bitcoin block (and not the stacks one)? If so, could it be the case that we end up writing a stacks block that is height-wise disconnected from the rest of the chain (and then maybe break something else)?

Reply:
Yeah, it's possible. This part of the test setup is non-intuitive and a little brittle. But the bigger issue is that we need to change how we write stacks blocks to the database (as part of https://github.com/stacks-network/sbtc/issues/559) so doing this "correctly" for these tests felt like wasted effort.